crates/biome_deserialize/src/lib.rs

```diff
@@ -1,31 +1,346 @@
+//! `biome_deserialize` provides a framework to deserialize textual data format.
+//!
+//! This is inspired by serde.
+//! 0ne of the main difference is the fault-tolerant behavior of `biome_deserialize`.
+//! Serde uses a fast-fail strategy, while `biome_deserialize` deserialize as much as possible
+//! and report several diagnostics (errors, warning, deprecation messages, ...).
+//!
+//! The two most important traits are [Deserializable] and [DeserializableValue].
+//!
+//!  - A type that implements `Deserializable` is a data structure that can be
+//!    deserialized from any supported data format
+//!  - A type that implements `DeserializableValue` is a data format that can
+//!    deserialize any supported data structure.
 mod diagnostics;
-mod visitor;
-
+mod impls;
 pub mod json;
 pub mod string_set;
-
 use biome_diagnostics::{Error, Severity};
-pub use diagnostics::{DeserializationAdvice, DeserializationDiagnostic};
+use biome_rowan::{TextRange, TokenText};
+pub use diagnostics::{DeserializationAdvice, DeserializationDiagnostic, ExpectedType};
+pub use impls::*;
 use std::fmt::Debug;
-pub use string_set::{deserialize_string_set, serialize_string_set, StringSet};
-pub use visitor::VisitNode;
+pub use string_set::StringSet;
+
+/// Implemented by data structures that can deserialize any [DeserializableValue].
+///
+/// `biome_deserialize` provides [Deserializable] implementations for common Rust types.
+/// To implement [Deserializable], you can reuse a type that implements [Deserializable] and
+/// turn the obtained value into what you want.
+///
+/// When deserializing more complex types, such as `struct`,
+/// you have to use a type that implements [DeserializationVisitor].
+///
+/// ### Example
```

Comment:
```suggestion
/// ## Example
```

The example is good, although we don't see how the implementation is eventually used, and this is true for the whole file. There isn't an example that shows how a "data format" is deserialized into the data structure defined in any example. 

I believe this is **very important**.